### PR TITLE
Support basic `hostInfo` command.

### LINKED ISSUE #157: Support `hostInfo`

### PR DESCRIPTION
**Pull request by ekalinin**
Closes #157 

Without `extra` field yet.

### CODE REVIEW

File: internal/handlers/handler_test.go [hunk 1]
```diff
@@ -558,6 +571,26 @@ func TestReadOnlyHandlers(t *testing.T) {
 			),
 		},
 
+		"HostInfo": {
+			req: types.MustMakeDocument(
+				"hostInfo", int32(1),
+			),
+			resp: types.MustMakeDocument(
+				"system", types.MustMakeDocument(
+					"currentTime", fmt.Sprintf("ISODate(%s)", nowStr),
```
Comment (AlekSi) [inline internal/handlers/handler_test.go:580]: I think that's really BSON time; ISODate is how it is represented in Mongo shell.

When you run FerretDB with `make run`, you should see the diff between FerretDB and MongoDB responses. That should make data types visible.

File: internal/handlers/handler_test.go [hunk 2]
```diff
@@ -558,6 +571,26 @@ func TestReadOnlyHandlers(t *testing.T) {
...
+					"hostname", hostname,
+					"cpuAddrSize", fmt.Sprintf("%d", strconv.IntSize),
+					"numCores", fmt.Sprintf("%d", runtime.NumCPU()),
+					"cpuArch", runtime.GOARCH,
+					"numaEnabled", "false",
```
Comment (AlekSi) [inline internal/handlers/handler_test.go:585]: I think that should be bool, not string. Again, diff output in `make run` should show that

Comment (ekalinin) [reply internal/handlers/handler_test.go]: Yep, fixes (as well as other fields). Thanks again.

File: internal/handlers/handler_test.go (hunk 1 above)
Comment (ekalinin) [reply internal/handlers/handler_test.go]: Got it, fixed. Thanks!

File: internal/handlers/handler_test.go [hunk 3]
```diff
@@ -413,6 +417,13 @@ func TestReadOnlyHandlers(t *testing.T) {
 		compareFunc func(t testing.TB, actual, expected any)
 	}
 
+	now := time.Now().UTC()
+	shared.Now = func() time.Time {
```
Comment (AlekSi) [inline internal/handlers/handler_test.go:421]: And we never set it back.

I think we should be fine with `testutil.CompareAndSetByPath` there, see this test: https://github.com/FerretDB/FerretDB/blob/630a7e9b336abaedb51ff6cd54b9219dbd4bee31/internal/handlers/handler_test.go#L554

File: internal/handlers/shared/msg_hostinfo.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			"system", types.MustMakeDocument(
+				"currentTime", now,
+				"hostname", hostname,
+				"cpuAddrSize", int32(strconv.IntSize),
+				"numCores", int32(runtime.NumCPU()),
+				"cpuArch", runtime.GOARCH,
+				"numaEnabled", false,
+			),
+			"os", types.MustMakeDocument(
+				"type", cases.Title(language.English).String(runtime.GOOS),
```
Comment (AlekSi) [inline internal/handlers/shared/msg_hostinfo.go:55]: Is `strings.Title` not enough?

Comment (ekalinin) [reply internal/handlers/shared/msg_hostinfo.go:55]: It's deprecated:
- https://github.com/golang/go/blob/7c94355b738170cf06484d502af7f2d935831c2b/src/strings/strings.go#L709

So I decided to use suggested replacement - `golang.org/x/text/cases`.

Comment (ekalinin) [reply internal/handlers/shared/msg_hostinfo.go:55]: But, of course, I can use `strings.Title` if you don't mind :)

File: internal/handlers/handler_test.go (hunk 3 above)
Comment (ekalinin) [reply internal/handlers/handler_test.go:421]: > And we never set it back.

Sure, I can reset the default value with `defer`.

> I think we should be fine with `testutil.CompareAndSetByPath` there, see this test

Could you, please, add more context here?
Should I remove this case into a separate test and use ` testutil.CompareAndSetByPath`?

Comment (ekalinin) [reply internal/handlers/handler_test.go]: Fixed. Thanks!

**Review by AlekSi — APPROVED** [review]

**Pull request merged**